### ISSUE TITLE
EurekaMonitors fails to unregister servo monitors from JmxMonitorRegistry

### ISSUE THREAD

**Issue by brenuart** [Netflix/eureka#638]
`EurekaMonitors.shutdown()` fails to properly unregister servo monitors from `JmxMonitorRegistry`. The following exception is thrown:
````
2015-09-14 20:28:59,904  WARN 44684 [       Thread-2]  --- com.netflix.servo.jmx.JmxMonitorRegistry           : Unable to un-register Monitor:MonitorConfig{name=RATE_LIMITED_FULL_FETCH_CANDIDATES, tags=class=EurekaMonitors, policy=DefaultPublishingPolicy}

javax.management.InstanceNotFoundException: com.netflix.servo:name=count,class=EurekaMonitors,id=RATE_LIMITED_FULL_FETCH_CANDIDATES,level=INFO,type=COUNTER
	at com.sun.jmx.interceptor.DefaultMBeanServerInterceptor.getMBean(DefaultMBeanServerInterceptor.java:1095)
	at com.sun.jmx.interceptor.DefaultMBeanServerInterceptor.exclusiveUnregisterMBean(DefaultMBeanServerInterceptor.java:427)
	at com.sun.jmx.interceptor.DefaultMBeanServerInterceptor.unregisterMBean(DefaultMBeanServerInterceptor.java:415)
	at com.sun.jmx.mbeanserver.JmxMBeanServer.unregisterMBean(JmxMBeanServer.java:546)
	at com.netflix.servo.jmx.JmxMonitorRegistry.unregister(JmxMonitorRegistry.java:125)
	at com.netflix.servo.DefaultMonitorRegistry.unregister(DefaultMonitorRegistry.java:143)
	at com.netflix.eureka.util.EurekaMonitors.shutdown(EurekaMonitors.java:184)
````

The `EurekaMonitors` class holds an enum for every monitor used by the Eureka sub-system. Problem is:
- registration is done with `enum.getName()` as name
- unregister is done with `enum.name()` as name (_not the same_)

**Comment by qiangdavidliu** [comment]
@brenuart thanks for the issue. Will fix.

**Comment by brenuart** [comment]
want a PR?

**Comment by qiangdavidliu** [comment]
I am doing some clean up work on the server side right now, so can just add the fix as part of that. Feel free to provide a PR as well.

**Comment by qiangdavidliu** [comment]
Resolved per #639
